woocommerce/woocommerce
# [Beta tester] Update "Reset Onboarding Wizard" tool to reset additional options

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:






Related to https://github.com/woocommerce/woocommerce/issues/52527

This PR enhances the WC Admin Test Helper tool by updating it to reset additional WooCommerce options, including woocommerce_allow_tracking, which was previously not reset.

Options documented at: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/docs/features/core-profiler.md

I didn't delete the `blogname` option as it's a core option and not a WooCommerce option and doesn't affect the WooCommerce setup. I also keep `woocommerce_task_list_tracked_completed_tasks` as I think it's useful to reset it. 




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally.
2. Run `cd to plugins/woocommerce-beta-tester` and run `pnpm run build:zip` to create a zip file.
3. Create a fresh site with WooCommerce installed.
4. Upload the beta tester zip file to the site and activate it.
5. Complete the setup wizard and opt-in to tracking.
6. Go to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` > Tools
7. Reset Onboarding Wizard
8. Confirm that all options specified in the update, including woocommerce_allow_tracking, woocommerce_default_country, and other core settings, are reset to their default values.

**Options that should be deleted:**

- `woocommerce_onboarding_profile`
- `woocommerce_task_list_tracked_completed_tasks`
- `woocommerce_private_link`
- `woocommerce_share_key`
- `woocommerce_store_pages_only`

**Options that should be reset to their default values:**

```js
const defaultOptions = {
  woocommerce_allow_tracking: 'no',
  woocommerce_default_country: 'US:CA',
  woocommerce_currency: 'USD',
  woocommerce_currency_pos: 'left',
  woocommerce_price_thousand_sep: ',',
  woocommerce_price_decimal_sep: '.',
  woocommerce_price_num_decimals: '2',
  woocommerce_coming_soon: 'no',
};
```



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
